pages/advanced-algorithms/available-algorithms/merge.mdx

```diff
@@ -0,0 +1,176 @@
+</Callout>
+
+### `node()`
+
+Provides a more flexible way of merging nodes than Cypher’s `MERGE` clause, by
+allowing labels and properties to be extracted as results of other procedures
+and sent as a list or map.
+
+{<h4> Input: </h4>}
+
+- `labels: List[string]` ➡ A list of all the labels to be added to the new node.
+- `identProps: Map` ➡ A map with key/value pairs used for identifying which nodes will be merged.
+- `createProps: Map` ➡ A map with key/value pairs which represent properties of the node, when the procedure creates a new node.
+Equivalent to using `ON CREATE SET` for properties when using `MERGE`.
+- `mergeProps: Map` ➡ A map with key/value pairs which represent properties of the node, when the procedure merges onto a node.
+Equivalent to using `ON MATCH SET` for properties when using `MERGE`.
+
+{<h4> Output: </h4>}
+
+- `node: Node` ➡ The node which was created or merged in the procedure.
+
+{<h4> Usage: </h4>}
+Consider a graph with two `Human`, and one `Alien` node.
+
+```cypher
+CREATE (h:Human {name:"Mark", age: 20}), (h2:Human {name:"Eva", age: 30}), (a:Alien {name: "Gorlock", age: 20});
+```
+Let's use `merge.node` on all our nodes, and add `galaxy` property on existing nodes. By leaving `labels` and `identProp` empty,
+all nodes in the graph will be affected by the procedure.
+
+```cypher
+CALL merge.node([],{},{},{galaxy: "Milky way"}) YIELD node RETURN node;
+```
+
+```plaintext
++----------------------------------------------------------------------------------------------------------------+
+| node                                                                                                           |
++----------------------------------------------------------------------------------------------------------------+
+| {"id": 0,"labels": ["Human"],"properties": {"age": 20,"galaxy": "Milky way","name": "Mark"},"type": "node"}    |
++----------------------------------------------------------------------------------------------------------------+
+| {"id": 1,"labels": ["Human"],"properties": {"age": 30,"galaxy": "Milky way","name": "Eva"},"type": "node"}     |
++----------------------------------------------------------------------------------------------------------------+
+| {"id": 2,"labels": ["Alien"],"properties": {"age": 20,"galaxy": "Milky way","name": "Gorlock"},"type": "node"} |
++----------------------------------------------------------------------------------------------------------------+
+```
+
+Now, let's use `merge.node` to either merge or create an `Alien` node with the name `Zumbi`. If that's a newly created node,
+`age` property will be 30, and if it's a merged node, `age` will be 40. Additionally, `galaxy` property will be used as an identifying property.
+NOTE: When using `merge.node` or `MERGE`, if there is no match in the graph, and a node is created, it will be created with identifying properties
+alongside create properties.
```

Comment:
```suggestion
Now, use the `merge.node` procedure to either merge or create an `Alien` node with the name `Zumbi`. If that's a newly created node, the
`age` property will be 30, and if it's a merged node, `age` will be 40. Additionally, the `galaxy` property will be used as an identifying property.

When using `merge.node` procedure or `MERGE` clause, if there is no match in the graph, and a node is created, it will be created with identifying properties
alongside create properties.
```

I don't understand this -  "it will be created with identifying properties alongside CREATE PROPERTIES? What do you mean with create properties?

Reply:
In the input section, I defined label, identProp, CreateProp, MergeProp. how merge works is it finds the nodes based on the label and identfying poperties, and sets MergeProperties on that node. If the node doesnt exist, it will create a new node, and set its properties to be the same like identProp and CreateProp(Create Properties). For example, label = Zumbi, identProp = {age =20), createProp = {galaxy: Andromeda}. If there is no node labeled Zumbi with age 20, it will be created, and it will also get property galaxy.

Reply:
i've added defined by the `createProps` input parameter.

Reply:
Great!